src/extension.ts

```diff
@@ -885,3 +887,33 @@ function isSupportedLanguage(languageId: TextDocument["languageId"]): boolean {
       return false;
   }
 }
+
+// NOTE(gabro): we would normally use the `configurationDefaults` contribution point in the
+// extension manifest but that's currently limited to language-scoped settings in VSCode.
+// We use this method to set global configuration settings such as `files.watcherExclude`
+// and `search.exclude`.
+function setupMetalsSettings() {
+  function updateFileConfig(
+    configKey: string,
+    propertyKey: string,
+    newValues: Record<string, boolean>
+  ) {
+    const config = workspace.getConfiguration(configKey);
+    const currentValues = (config.get(propertyKey) || {}) as Record<
+      string,
+      boolean
+    >;
+    config.update(
+      propertyKey,
+      { ...currentValues, ...newValues },
+      ConfigurationTarget.Global
+    );
+  }
+  const ignoredMetalsDirs = {
+    "**/.bloop": true,
+    "**/.metals": true,
+    "**/target": true,
+  };
+  updateFileConfig("files", "watcherExclude", ignoredMetalsDirs);
+  updateFileConfig("search", "exclude", ignoredMetalsDirs);
```

Comment:
I've limited it to `files.watcherExclude` (avoid file watching, for performance) and `search.exclude` (avoid showing generated files in search results).

I've not added `files.exclude` because that's a broader settings that would also hide those dirs from the file explorer, which I don't think is a good default.

Reply:
Good idea! I was worried this could mean `.bloop` would just disappear and cause additional confusion. I had a problem like that previously and I did not fully understood what happened for a while.